### PR TITLE
Step9 and 10 pom and suggested selectors

### PR DESCRIPTION
**Pull request by GirlesaQuinteroR**
I tried everything, but this one was necessary 👍 
image

### CODE REVIEW

File: cypress/integration/buy-shirt-copy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+describe("Buy a t-shirt", () => {
```
Comment (holgiosalos) [inline cypress/integration/buy-shirt-copy.spec.ts]: Remove this file since it is a copy

File: cypress/page/login.page.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    private signInBtn: string
+
+    constructor() {
+        this.email = "#email"
+        this.password = "#passwd"
+        this.signInBtn = "#SubmitLogin"
+    }
+
+    public typeUser(): void {
+        cy.get(this.email).type("aperdomobo@gmail.com")
```
Comment (holgiosalos) [inline cypress/page/login.page.ts]: receive the email as param of the method

File: cypress/page/login.page.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    public typePassword(): void {
+        cy.get(this.password).type("WorkshopProtractor")
```
Comment (holgiosalos) [inline cypress/page/login.page.ts]: receive the pass as param of the method

File: cypress/page/login.page.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    public submitLoginCredentials(): void {
+        cy.get(this.signInBtn).click()
+    }
+}
+export { LoginPage }
```
Comment (holgiosalos) [inline cypress/page/login.page.ts]: EOF

File: cypress/page/payment-step.page.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public selectPaymentMethod(): void {
+        cy.get(this.paymentMethod).click()
+    }
+
+    public confirmTheOrder(): void {
+        cy.get(this.confirmOrder).contains("I confirm my order").click()
+    }
+
+    public confirmationPaymentMessage(): void {
+        cy.get(this.confirmationMessage).contains("Your order on My Store is complete.")
```
Comment (holgiosalos) [inline cypress/page/payment-step.page.ts]: receive the expected message as input param of the method, also rename the method to be validateConfirmationPaymentMessage or something like that, in order to indicate we are asserting or validating things here

File: cypress/page/products-list.page.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    constructor() {
+        /*With this selector suggested is possible to have the number 1 as a variable
+        to tests more items in the page*/
+        this.tShirtMenu = "[title='Add to cart'][data-id-product='1']";
+        this.proceedToCheckout = ".btn[title='Proceed to checkout']"
+    }
+
+    public goToTShirtMenu(): void {
+        cy.get(this.tShirtMenu).click({force: true, multiple: true})
```
Comment (holgiosalos) [inline cypress/page/products-list.page.ts]: why it is needed the `{force: true, multiple: true}`

File: cypress/page/shipping-step.page.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    public processCarrier(): void {
+        cy.get(this.proceedToCheckout).click()
+    }
+
+    public termsAndConditionsBox(): void {
+        cy.get(this.termsAndConditions).click()
+    }
+}
+export { ShippingCartPage }
```
Comment (holgiosalos) [inline cypress/page/shipping-step.page.ts]: EOF

File: cypress/integration/buy-shirt.spec.ts [hunk 8]
```diff
@@ -1,25 +1,38 @@
...
-      cy.get(".cart_navigation > .button > span").click()
-      cy.get("#cgv").click()
-      cy.get(".button[name='processCarrier']").click()
-      cy.get(".bankwire").click()
-      cy.get(".button[type='submit']").contains("I confirm my order").click()
-      cy.get("#center_column > div > p > strong")
-      .should("have.text", "Your order on My Store is complete.")
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage()
+    productsListPage.goToTShirtMenu()
```
Comment (holgiosalos) [inline cypress/integration/buy-shirt.spec.ts]: The method `goToTShirtMenu()` should go on the `menuContentPage` page object

File: cypress/page/index.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export { MenuContentPage } from "./menu-content.page"
+export { ShoppingCartPage } from "./shoping-cart.page"
+export { ShippingCartPage } from "./shipping-step.page"
+export { LoginPage } from "./login.page"
+export { PaymentStep } from "./payment-step.page"
+export { AdressStep } from "./address-step.page"
+export { ProductsListPage } from "./products-list.page"
+
```
Comment (holgiosalos) [inline cypress/page/index.ts]: Remove this extra blank line

File: cypress/page/menu-content.page.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+class MenuContentPage {
+    private menuContentPageURL: string
+
+    constructor() {
+        this.menuContentPageURL = "http://automationpractice.com/"
```
Comment (holgiosalos) [inline cypress/page/menu-content.page.ts]: this MenuContentPage should remain as it is in the guide:
image

File: cypress/page/products-list.page.ts (hunk 6 above)
Comment (holgiosalos) [reply cypress/page/products-list.page.ts]: this method should be in the MenuContentPage class

File: cypress/page/products-list.page.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+class ProductsListPage {
+    private tShirtMenu: string;
+    private proceedToCheckout: string
+
+    constructor() {
+        /*With this selector suggested is possible to have the number 1 as a variable
+        to tests more items in the page*/
+        this.tShirtMenu = "[title='Add to cart'][data-id-product='1']";
```
Comment (holgiosalos) [inline cypress/page/products-list.page.ts]: this selector does not belong to the t-shirt Menu, it is pointing to the add to cart button. I expect this Page Object to have two properties, one for the AddToCart button and other for the ProceedToCheckout button. You have the proceed to checkout but you are using the AddToCart selector for the tshirtMenu option that should go on the MenuContentPage class

**Comment by holgiosalos** [comment]
> I tried everything, but this one was necessary 👍
> image

You need this because there is a missing step in previous page objects

File: cypress/integration/buy-shirt-copy.spec.ts (hunk 1 above)
Comment (GirlesaQuinteroR) [reply cypress/integration/buy-shirt-copy.spec.ts]: Removed

File: cypress/page/login.page.ts (hunk 3 above)
Comment (GirlesaQuinteroR) [reply cypress/page/login.page.ts]: Done

File: cypress/page/login.page.ts (hunk 2 above)
Comment (GirlesaQuinteroR) [reply cypress/page/login.page.ts]: Done

File: cypress/page/payment-step.page.ts (hunk 5 above)
Comment (GirlesaQuinteroR) [reply cypress/page/payment-step.page.ts]: Got it! Validate and verify, the QA's verbs. At this time, Verify the "acceptance criteria"

File: cypress/integration/buy-shirt.spec.ts (hunk 8 above)
Comment (GirlesaQuinteroR) [reply cypress/integration/buy-shirt.spec.ts]: Done

File: cypress/page/menu-content.page.ts (hunk 10 above)
Comment (GirlesaQuinteroR) [reply cypress/page/menu-content.page.ts]: Just to know, Is there any special reason to have it like the guide? Maybe further I will need the specific CSS or something?

File: cypress/page/login.page.ts (hunk 4 above)
Comment (GirlesaQuinteroR) [reply cypress/page/login.page.ts]: Saved it with the settings now

File: cypress/page/shipping-step.page.ts (hunk 7 above)
Comment (GirlesaQuinteroR) [reply cypress/page/shipping-step.page.ts]: Saved it with the settings now

File: cypress/page/index.ts (hunk 9 above)
Comment (GirlesaQuinteroR) [reply cypress/page/index.ts]: Done.

File: cypress/page/products-list.page.ts (hunk 11 above)
Comment (GirlesaQuinteroR) [reply cypress/page/products-list.page.ts]: Got it! Also, it solved the issue with the double `goToPayment().`

File: cypress/page/products-list.page.ts (hunk 6 above)
Comment (GirlesaQuinteroR) [reply cypress/page/products-list.page.ts]: Resolved

**Review by holgiosalos — APPROVED** [review]
Good job with the selectors!

**Pull request merged**